### PR TITLE
Backend/create functions

### PR DESCRIPTION
**Pull request by markc331**
Create functions for user account and task creation 
User Account is made using user name, and password, and an ID is generated for that account.

Tasks Table is made using info given by the user and uses the connection file to grab the corresponding account id and link the table to the account

### CODE REVIEW

File: backend/connect.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    - add method of verification by matching credentials to entries on the table 
+        (i.e name and password searched within the table to obtain the 
+        corresponding account id)
+        (could include more unique identifiers like email, unique username, etc.)
+    
+**************************************************************************
+<?php
+    //connects to database
+    $conn = mysqli_connect('localhost', 'root', '9204Alex', 'coen146');
```
Comment (loshprung) [inline backend/connect.php]: Connection should call to variables set in a config file (do not leave these with your test credentials)

File: backend/connect.php [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    //check connection
+    if(!$conn){
+        echo 'Connection Error:' . mysqli_connect_error();
+    }
+
+    session_start();
+
+    $sql = 'SELECT account_id FROM accounts';
```
Comment (loshprung) [inline backend/connect.php]: Do a 'WHERE' clause instead of getting every account id

File: backend/connect.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    $result = mysqli_query($conn, $sql);
+
+    if(mysqli_num_rows($result) > 0)
+    {
+        while($row = mysqli_fetch_assoc($result)){
+            $_SESSION['account_id']= $row['account_id'];
```
Comment (loshprung) [inline backend/connect.php]: Use '$_REQUEST' instead of '$_SESSION' throughout

File: backend/create_account_table.sql [hunk 4]
```diff
@@ -1,6 +1,6 @@
 CREATE TABLE Accounts(
```
Comment (loshprung) [inline backend/create_account_table.sql]: I would be interested in debating the choice to change id to account_id in the Accounts table. I would argue in favor of keeping the column 'id'. Why the change?

File: backend/new_account.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+//change password and database accordingling
+$conn = mysqli_connect('localhost', 'root', 'password', 'database');
+
+//check connection
+if(!$conn){
+    echo 'Connection Error:' . mysqli_connect_error();
+}
+
+
+$name = $password = '';
```
Comment (loshprung) [inline backend/new_account.php:8]: Make whatever this is doing less cryptic

File: backend/new_account.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        $name = mysqli_real_escape_string($conn, $_POST['name']);
+        $password = mysqli_real_escape_string($conn, $_POST['password']);
+
+        $sql = "INSERT INTO accounts(name,password) Values('$name', '$password')";
+
+        //save to db and check
+        if(mysqli_query($conn, $sql)){
+            //success
+            //if false (no errors) redirects to new submission page 
+            header('Location:new_account.php');
```
Comment (loshprung) [inline backend/new_account.php:45]: Do not redirect

File: backend/new_task.php [hunk 7]
```diff
@@ -0,0 +1,92 @@
+**************************************************************************
+    Program takes in argument from the front-end interface along with the corresponding account_id 
+    obtained from the db_connect file and stores the entries to the corresponding columns in the 
+    tasks table within the server
+    
+**************************************************************************
+
+<?php 
+
+session_start();
```
Comment (loshprung) [inline backend/new_task.php]: What is the point of this? How come it is in new_task.php, but not new_account.php?

File: backend/new_task.php [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        $deadline = mysqli_real_escape_string($conn, $_POST['deadline']);
+        $priority = mysqli_real_escape_string($conn, $_POST['priority']);
+
+        $sql = "INSERT INTO tasks(account_id, tasks_name,category,deadline,priority) Values('$account_id','$tasks_name', '$category', '$deadline', '$priority')";
+
+        //save to db and check
+        if(mysqli_query($conn, $sql)){
+            //success
+            //if false (no errors) redirects to new submission page 
+            header('Location:create_task.php');
```
Comment (loshprung) [inline backend/new_task.php]: Do not redirect

**Comment by loshprung** [comment]
This branch has conflicts. Once it is approved, I can manually merge it and handle the conflicts

File: backend/new_task.php [hunk 9]
```diff
@@ -7,57 +7,50 @@
...
+    include('connect.php');
 
-//may need to use sessions to connect to specific account
 
-$tasks_name = $category = '';
-$deadline = '0000-00-00';
-$priority = 0;
+    $tasks_name = $category = '';
+    $deadline = '0000-00-00';
+    $priority = 0;
```
Comment (loshprung) [inline backend/new_task.php:22]: We may want to make these columns allowed to be NULL in the future...

File: backend/create_account_table.sql [hunk 10]
```diff
@@ -1,6 +1,6 @@
...
-	id INT UNSIGNED NOT NULL AUTO_INCREMENT,
+	account_id INT UNSIGNED NOT NULL AUTO_INCREMENT,
 	name VARCHAR(256) NOT NULL UNIQUE,
 	password VARCHAR(32) NOT NULL,
-	PRIMARY KEY(id)
+	PRIMARY KEY(account_id)
 );
```
Comment (loshprung) [inline backend/create_account_table.sql]: Changing the column name has pretty major repercussions and will make merging difficult. Please revert the changes made to this file and related changes

**Review by loshprung — APPROVED** [review]
I think this should be good, and there are no merge conflicts anymore. I may do some additional consolidation in a future commit

**Review by maddiewaldie — APPROVED** [review]
Looks good to me!
